lua/neogit/lib/git/rebase.lua

```diff
@@ -58,6 +59,56 @@ function M.onto(start, newbase, args)
   end
 end
 
+function M.reword(commit, message, args)
+  -- based on https://peterevans.dev/posts/how-to-rewrite-git-commit-messages-non-interactively/
+  -- TODO: Error handling
+  local result = process
+    .new({
+      cmd = {
+        "git",
+        "commit",
+        "--allow-empty",
+        "--only",
+        "-m",
+        "amend! " .. commit .. "\n\n" .. message,
+      },
+      on_error = function()
+        return false
+      end,
+    })
+    :spawn_blocking()
```

Comment:
I think something like this would work:
`cli.commit.allow_empty.only.message(("amend! %s\n\n%s"):format(commit, message)).call_sync()`

I don't know if all of those fields are defined in the git.cli lib, but thats not too hard. Look for `commit = config { ... }`. I'm on mobile, so if that doesn't actually exist I'll add it tonight.

Reply:
Ahh okay I see. This is a method chaining API 🤦 Think I could solve this now. Thanks

Reply:
Using the `git.cli...` now